quarkus/tests/integration/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java

```diff
@@ -100,7 +103,34 @@ public void beforeEach(ExtensionContext context) throws Exception {
             onBeforeStartDistribution(context.getRequiredTestMethod().getAnnotation(BeforeStartDistribution.class));
 
             if (launch != null) {
-                result = dist.run(Arrays.asList(launch.value()));
+                List<String> argsFromLaunchAnnotation = new ArrayList<>(Arrays.asList(launch.value()));
+
+                WithLegacyStoreOnly withLegacyStoreOnly = context.getRequiredTestMethod().getAnnotation(WithLegacyStoreOnly.class);
+                if (withLegacyStoreOnly == null) {
+                    withLegacyStoreOnly = context.getTestClass()
+                            .orElse(Object.class)
+                            .getDeclaredAnnotation(WithLegacyStoreOnly.class);
+                }
+
+                Boolean useCHMStorageWhenPossible = Boolean.parseBoolean(System.getProperty("kc.test.storage.use-chm-storage-when-possible", "true"));
+                if (useCHMStorageWhenPossible &&
+                    withLegacyStoreOnly == null &&
+                    CollectionUtil.isNotEmpty(argsFromLaunchAnnotation)) {
+
+                    argsFromLaunchAnnotation.add(1, "--storage=chm");
+
+                    AtomicReference<String> featuresOptionFound = new AtomicReference<>(null);
+                    argsFromLaunchAnnotation.stream()
+                                            .filter(oneOption -> oneOption.startsWith("--features="))
+                                            .findFirst()
+                                            .ifPresent(oneOption -> featuresOptionFound.set(oneOption));
+                    if (featuresOptionFound.get() != null) {
```

Comment:
The `--features` param should be added to the args also when no other features are specified, right? The thing is, that the Map Storage feature is still in an Experimental state and it needs to be turned on.

Reply:
@mabartos,

I noted that, If we set the `--storage=chm` and even without adding something like `--feature=map-storage` flag, the Experimental `map-storage` is automatically activated. This is what we can see at console log:

`... WARN  [org.keycloak.common.Profile] (main) Experimental features enabled: map-storage`

But If I set the `--features=...` flag without any mention to `map-storage`, then one Exception is raised because the `map-storage` is not activated.

Therefore, I'm gonna keep this code as it is. WDYT?